src/PrestaShopBundle/Resources/config/services/bundle/controller.yml

```diff
@@ -142,5 +142,14 @@ services:
       multistoreFeature: "@prestashop.adapter.multistore_feature"
       multistoreContext: "@prestashop.adapter.shop.context"
       entityManager: "@doctrine.orm.entity_manager"
+      productRepository: '@PrestaShop\PrestaShop\Adapter\Product\Repository\ProductRepository'
```

Comment:
This one also uses public properties, but it's a flaw that was already present and we'll fix this in v9.0 (long story short this shouldn't be a controller but a rendering component)